frontend/src/components/Post.js

```diff
@@ -6,22 +6,55 @@ import './Post.css'
 const Post = props => {
   const [replyOpen, setReplyOpen] = React.useState(false)
 
-  const toggleReply = () => setReplyOpen(!replyOpen)
+  const toggleReply = () => {
+    setReplyOpen(!replyOpen)
+  }
 
   const saveComment = commentData => {
     setReplyOpen(false)
     props.onComment(props.post._id, commentData)
   }
 
+  const originalVotes = props.post.upVotes - props.post.downVotes;
+
+  //upvote downvote & toggle
+  const [newVotes, setNV] = React.useState(originalVotes);
+
+  const clickUV = () => {
+    
+    const isUpVoted = (newVotes > originalVotes);
+    if (isUpVoted) {
+      setNV(originalVotes)
+      props.removeUV()
+    } else {
```

Comment:
Totally understand that you missed this, but there's an edge case this misses for keeping track of total upvotes and downvotes: if the post is currently "downvoted" and you click "upvote," the total upvotes counter will increase by 1, but the total downvotes counter won't decrease!

To fix this, I'd add an extra `if` statement: 
```js
else if (newVotes < originalVotes) { 
  setNewVotes(originalVotes + 1)
  props.onDownvote(false) // reduce total downvotes by 1
  props.onUpvote(true) // increase total upvotes by 1
}
```